src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -1,52 +1,186 @@
+                return;
+            }
+
+            ResolveManagedProfilerDirectory();
+            StartupLogger.Log($"Will try to load {_assemblyNames.Length} assemblies. Directory: \"{s_managedProfilerDirectory}\".");
 
             try
             {
-                AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve_ManagedProfilerDependencies;
+                AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolveEventHandler;
             }
             catch (Exception ex)
             {
                 StartupLogger.Log(ex, "Unable to register a callback to the CurrentDomain.AssemblyResolve event.");
             }
 
-            TryLoadManagedAssembly();
+            for (int i = 0; i < _assemblyNames.Length; i++)
+            {
+                string assemblyName = _assemblyNames[i];
+
+                try
+                {
+                    LoadAndStartAssembly(assemblyName);
+                }
+                catch (Exception ex)
+                {
+                    StartupLogger.Log(ex, $"Error loading or starting a managed assembly (\"{assemblyName}\").");
+                }
+            }
         }
 
-        internal static string ManagedProfilerDirectory { get; }
+        private static void LoadAndStartAssembly(string assemblyName)
+        {
+            if (assemblyName == null)
+            {
+                StartupLogger.Log($"Skipping loading assembly because the specified {nameof(assemblyName)} is null.");
+                return;
+            }
+
+            assemblyName = assemblyName.Trim();
+
+            if (assemblyName.Length == 0)
+            {
+                StartupLogger.Log($"Skipping loading assembly because the specified {nameof(assemblyName)} is \"\".");
+                return;
+            }
 
-        private static void TryLoadManagedAssembly()
+            StartupLogger.Log($"Loading managed assembly \"{assemblyName}\""
+                            + $" and invoking method \"{TargetLibraryEntrypointMethod}\""
+                            + $" in type \"{TargetLibraryEntrypointFullTypeName}\".");
+
+            Assembly assembly = Assembly.Load(assemblyName);
+            if (assembly == null)
+            {
+                StartupLogger.Log($"Could not load managed assembly \"{assemblyName}\".");
+                return;
+            }
+
+            Type entryPointType = assembly.GetType(TargetLibraryEntrypointFullTypeName, throwOnError: false);
+            if (entryPointType == null)
+            {
+                StartupLogger.Log($"Could not obtain type \"{TargetLibraryEntrypointFullTypeName}\""
+                                + $" from managed assembly \"{assemblyName}\".");
+                return;
+            }
+
+            MethodInfo entryPointMethod = entryPointType.GetRuntimeMethod(TargetLibraryEntrypointMethod, parameters: new Type[0]);
```

Comment:
Silly thing, but `new Type[0]` can be static so we can reuse it in multiple assembly loads.

Reply:
Given how few times this code runs per process, I kinda tend to go towards less declared statics => better overview/readability. 

Good point, thought. I keep it in mindas I clean things up.